pkg/config/config.go

```diff
@@ -402,6 +402,10 @@ func (dc *DeploymentConfig) validateConfig() {
 	if err != nil {
 		log.Fatal(err)
 	}
+	err = dc.Config.checkBlueprintName()
```

Comment:
I feel that everything that will eventually become a label (blueprint name, deployment name, etc) should run this test. Should you not generalize to check label requirements? and then run it for both blueprint name and deployment name?